### DISCUSSION TITLE
Post build actions

### DISCUSSION THREAD

**Discussion by ante-maric** [renovatebot/renovate#23330] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.7

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Server

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I have some nuget packages that have post build actions that copy files to a current repository. So, before commit is created I need to build my dotnet solution for those actions to be executed and then new/updated files included in the commit.

### What you tried so far.

I was looking for a way to install dotnet and run dotnet build, but didn't find a way to do this.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
Tried postUpgradeTasks?

**Reply by ante-maric** [reply]
Yes, but dotnet is not available in the container. I tried, before "dotnet build" command to execute "apt-get update && apt-get install -y dotnet-sdk-6.0", but that also did not work. At the moment I don't have the error message.

**Reply by rarkins** [reply]
Try running "install-tool sorbet x.y.z" first (pick a suitable version)

**Reply by ante-maric** [reply]
Thank you, I did manage to do it by using `install-tool dotnet x.y.z`. But the down side is that nothing gets persisted so it has to install it each time it runs.

**Reply by viceice** [reply, marked as answer]
you can extend the default renovate image with your tools, so they are already preinstalled

**Reply by ante-maric** [reply]
That is exactly what I end up doing.
